README.md

```diff
@@ -17,10 +17,54 @@ npm start
 ### Testing
 
 - `npm test` - run unit and feature tests with Jest
-- `npm run cypress` - run end-to-end tests with Cypress against local dev server running in separate terminal
+- `npm run cypress` - open the [Cypress](https://docs.cypress.io/guides/overview/why-cypress.html) end-to-end test
+  runner (local dev server must be running in separate terminal)
+- `npm run cypress:run` - run end-to-end tests once (local dev server must be running in separate terminal)
 - `npm run storybook` - manually test components in isolation in
   [Storybook](https://storybook.js.org/docs/react/get-started/introduction), at http://localhost:6006
 
+#### Visual regression
+
+Cypress is used for end-to-end testing, but also for visual regression testing. The idea is to take screenshots
+(referred to as "snapshots") of the app in various situations and compare them with previously approved "reference
+snapshots". If any pixel has changed, the test fails and a diff image that highlights the differences is created.
+
+Cypress is unable to take consistent screenshots across platforms because the exact rendering depends on the GPU (even
+when Docker is involved). For this reason, visual regression tests must always run on the same platform, that is, on the
+CI. Environment variable `CYPRESS_TAKE_SCREENSHOTS` is used to enable visual regression testing in the CI. This variable
+is set automatically in the `cypress.yml` GitHub workflow, which runs on push to any branch.
```

Comment:
It's important to check for regressions before merging to master. In my opinion, it should remain part of a PR review process.

Reply:
I would rather find ways to accelerate the CI workflow, with caching or by building the app without production optimisations for instance.

Reply:
> It's important to check for regressions before merging to master. In my opinion, it should remain part of a PR review process.

Yeah but you could imagine run the job only when there is a PR for the branch rather than for any branch.